python/metatensor-learn/metatensor/learn/data/dataset.py

```diff
@@ -70,6 +70,13 @@ def __len__(self) -> int:
         """
         return self._size
 
+    def __iter__(self):
```

Comment:
so this will not work in torch script, but I think it's fine

Reply:
Ah interesting